### PR TITLE
Add base set of labels

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
Info: https://github.com/metarhia/Metarhia/issues/37.

### CODE REVIEW

File: labels.json [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+    "color": "5319e7",
+    "description": "Issues and PRs related to the performance of Node.js."
+  },
+  {
+    "name": "question",
+    "color": "cc317c",
+    "description": "Issues that look for answers."
+  },
+  {
+    "name": "refactor to ES6+",
```
Comment (nechaido) [inline labels.json]:
```suggestion
    "name": "refactoring",
```

File: labels.json [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+    "color": "9944dd",
+    "description": "Issues and PRs related to the Windows platform."
+  },
+  {
+    "name": "wontfix",
+    "color": "ededed",
+    "description": "Issues that will not be fixed, e.g. due to backwards compatibility concerns."
+  },
+  {
+    "name": "work in progress",
```
Comment (nechaido) [inline labels.json]: There is already `in progress`, I'd prefer to only have one label for this purpose.

File: labels.json [hunk 3]
```diff
@@ -0,0 +1,302 @@
...
+    "name": "landed",
+    "color": "6f42c1",
+    "description": "PRs that were landed."
+  },
+  {
+    "name": "lib",
+    "color": "fad8c7",
+    "description": "Issues and PRs related to general changes in the lib directory."
+  },
+  {
```
Comment (lundibundi) [inline labels.json]: I don't this we need: `lint`, `lts`, `misspell`, `npm`, `optimization`, `server`, `src`, `triage-*`.

File: labels.json [hunk 4]
```diff
@@ -0,0 +1,257 @@
...
+    "color": "006b75",
+    "description": "Issues and PRs related to the documentations."
+  },
+  {
+    "name": "duplicate",
+    "color": "ededed",
+    "description": "Issues and PRs that are duplicates of other issues or PRs."
+  },
+  {
+    "name": "experimental",
```
Comment (nechaido) [inline labels.json]: Ditto.

File: labels.json (hunk 2 above)
Comment (nechaido) [reply labels.json]: I'd prefer to have `in progress` to show that issue\PR is being worked on, we have Drafts for WIP: PR, that are still not finished.

File: labels.json [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+  },
+  {
+    "name": "help wanted",
+    "color": "159818",
+    "description": "Issues that need assistance from volunteers or PRs that need help to proceed."
+  },
+  {
+    "name": "in progress",
+    "color": "d93f0b",
+    "description": "Issues and PRs that are still in progress."
```
Comment (nechaido) [inline labels.json]:
```suggestion
    "description": "Issues and PRs that are in progress."
```

**Comment by SemenchenkoVitaliy** [comment]
ping @nechaido @belochub.

**Comment by belochub** [comment]
@tshemsedinov can you land this, please?

File: labels/make-preview.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+          if (C[i] <= 0.03928) {
+            C[i] = C[i] / 12.92;
+          } else {
+            C[i] = Math.pow((C[i] + 0.055) / 1.055, 2.4);
+          }
+        }
+
+        const L = 0.2126 * C[0] + 0.7152 * C[1] + 0.0722 * C[2];
+
+        if (L > 0.179) {
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
`return iL > 0.179 ? 'black' : 'white';`
```

File: labels/make-preview.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+      const labels = ${JSON.stringify(labels)};
+
+      function calculateColor(bgColor) {
+        const R = parseInt(bgColor.slice(0, 2), 16),
+          G = parseInt(bgColor.slice(2, 4), 16),
+          B = parseInt(bgColor.slice(4, 6), 16);
+
+        const C = [R / 255, G / 255, B / 255];
+
+        for (var i = 0; i < C.length; ++i) {
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        for (let i = 0; i < c.length; ++i) {
```

File: labels/make-preview.js [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+        display: inline-block;
+      }
+    </style>
+  </head>
+  <body>
+    <script>
+      const labels = ${JSON.stringify(labels)};
+
+      function calculateColor(bgColor) {
+        const R = parseInt(bgColor.slice(0, 2), 16),
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        const r = parseInt(bgColor.slice(0, 2), 16);
```

File: labels/make-preview.js [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+          G = parseInt(bgColor.slice(2, 4), 16),
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        const g = parseInt(bgColor.slice(2, 4), 16);
```

File: labels/make-preview.js [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+          B = parseInt(bgColor.slice(4, 6), 16);
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        const b = parseInt(bgColor.slice(4, 6), 16);
```

File: labels/make-preview.js [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+        const C = [R / 255, G / 255, B / 255];
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        const c = [r / 255, g / 255, b / 255];
```

File: labels/make-preview.js [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+        for (var i = 0; i < C.length; ++i) {
+          if (C[i] <= 0.03928) {
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
          if (c[i] <= 0.03928) {
```

File: labels/make-preview.js [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+            C[i] = C[i] / 12.92;
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
            c[i] = c[i] / 12.92;
```

File: labels/make-preview.js [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+          } else {
+            C[i] = Math.pow((C[i] + 0.055) / 1.055, 2.4);
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
            c[i] = Math.pow((c[i] + 0.055) / 1.055, 2.4);
```

File: labels/make-preview.js [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+          }
+        }
+
+        const L = 0.2126 * C[0] + 0.7152 * C[1] + 0.0722 * C[2];
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
        const l = 0.2126 * c[0] + 0.7152 * c[1] + 0.0722 * c[2];
```

File: labels/make-preview.js [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+        if (L > 0.179) {
+          return 'black';
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
```

File: labels/make-preview.js [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+        } else {
+          return 'white';
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
```

File: labels/make-preview.js [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+        }
```
Comment (tshemsedinov) [inline labels/make-preview.js]:
```suggestion
```

**Comment by tshemsedinov** [comment]
Landed in https://github.com/metarhia/Metarhia/commit/cc15523ab6c7deaa621e67ac2b8e8215438ef522 and https://github.com/metarhia/Metarhia/commit/c87a3ea53006d447ff4862e449b1a3653258e4e3

**Pull request closed without merging**